xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/resources/classes/DBListClassPropertyValuesProvider.java

```diff
@@ -61,4 +64,26 @@ protected PropertyValues getAllowedValues(DBListClass dbListClass, int limit, St
             () -> getValues(this.allowedValuesQueryBuilder.build(dbListClass), limit, filter, dbListClass),
             dbListClass.getOwnerDocument().getAuthorReference(), dbListClass.getDocumentReference());
     }
+
+    @Override
+    public PropertyValue getValue(ClassPropertyReference propertyReference, Object rawValue) throws XWikiRestException
+    {
+        PropertyValue propertyValue = null;
+        DBListClass dbListClass = getPropertyDefinition(propertyReference);
+        if (rawValue != null) {
+            try {
+                propertyValue = this.authorExecutor
+                        .call(() -> getValue(this.allowedValuesQueryBuilder.build(dbListClass), rawValue.toString(),
+                                dbListClass), dbListClass.getOwnerDocument().getAuthorReference(),
+                                dbListClass.getDocumentReference());
+            } catch (Exception e) {
+                throw new XWikiRestException(e);
+            }
+        }
+        if (propertyValue == null) {
+            propertyValue = super.getValue(propertyReference, rawValue);
+        }
+
+        return propertyValue;
```

Comment:
I find this overly complicated. Why not simply call

    getValues(propertyReference, 1, rawValue.toString())

and return the first value from the returned ``PropertyValues``?

Reply:
@mflorea The query in `#getValues` uses this text filter: `text like '%value%'`, hence the first value returned by `#getValues` might not be the one that we are looking for. What we want is `text = 'value'`. (or `text like 'value'`)

Reply:
@atallahade right, it looks good then.